lib/akamod/select-id.py

```diff
@@ -34,25 +32,33 @@ def selid(body, ctype, prop='handle'):
     request_headers = copy_headers_to_dict(request.environ)
     source_name = request_headers.get('Source')
 
-    id = None
-    if exists(data,prop):
-        v = getprop(data,prop)
-        if isinstance(v,basestring):
-            id = v
+    record_id = None
+    if exists(data, prop):
+        v = getprop(data, prop)
+        if isinstance(v, basestring):
+            record_id = v
         else:
             if v:
                 for h in (v if isinstance(v, list) else [v]):
                     if is_absolute(h):
-                        id = h
-                if not id:
-                    id = v[0]
+                        record_id = h
+                if not record_id:
+                    record_id = v[0]
 
-    if not id:
+    if not record_id:
         response.code = 500
         response.add_header('content-type', 'text/plain')
         return "No id property was found"
 
-    data[u'_id'] = COUCH_REC_ID_BUILDER(source_name, id)
-    data[u'id']  = hashlib.md5(data[u'_id']).hexdigest()
+    '''
+    If the useSource parameter is True (default) than prepend it to
+    the id and use that value when hashing for the DPLA id
+    '''
+    if use_source.lower() == 'yes':
+        data[u'_id'] = couch_rec_id_builder(source_name, record_id)
```

Comment:
It looks like the second parameter to `utilities.couch_rec_id_builder()` is supposed to be a `dict`, but it appears from the way `record_id` is assigned above (and the way it's named) that it's going to be a string. Can you verify / clarify?